module/VuFind/src/VuFind/ILS/Driver/MultiBackend.php

```diff
@@ -190,20 +190,53 @@ public function getStatus($id)
      */
     public function getStatuses($ids)
     {
-        $items = [];
+        // Group records by source and request statuses from the drivers
+        $grouped = [];
         foreach ($ids as $id) {
-            try {
-                $items[] = $this->getStatus($id);
-            } catch (ILSException $e) {
-                $items[] = [
-                    [
-                        'id' => $id,
-                        'error' => 'An error has occurred'
-                    ]
+            $source = $this->getSource($id);
+            if (!isset($grouped[$source])) {
+                $driver = $this->getDriver($source);
+                $grouped[$source] = [
+                    'driver' => $driver,
+                    'ids' => []
                 ];
             }
+            $grouped[$source]['ids'][] = $id;
+        }
+
+        // Process each group
+        $results = [];
+        foreach ($grouped as $source => $current) {
+            if ($current['driver']) {
```

Comment:
What are the circumstances under which `$current['driver']` would be non-truthy here? Is skipping the records the best approach to a missing driver, or is that more of a "throw exception" sort of scenario? If exception throwing is appropriate, would it be better to do that right after the getDriver call?

Reply:
That's the case at least when a driver is not configured for the given source. That's valid situation and shouldn't throw an exception. The old version also silently returned empty results for those.

Reply:
Okay, great, that makes sense. Might be worth adding a comment here, like:

`// If no driver configured for this source, do nothing`

(just to confirm for future reference that the behavior is intentional, in case I forget about this conversation when reviewing the code in the future).